### PR TITLE
Move Config Defaults Out of Python

### PR DESCRIPTION
**Pull request by alecpm**
## Description
This PR implements [ScrumDo Story 494](https://app.scrumdo.com/projects/story_permalink/1829486). It moves default config values from calls like `config.get(key, default)` to the `global_config_defaults.txt`. It also implements automatic whitespace stripping of strings in `config.get` and fixes some issues with how config was being manipulated in tests. Custom configuration in demo experiments has also been adapted to set default values in `config.txt` instead of in the experiment code.

## Motivation and Context
See [ScrumDo Story 494](https://app.scrumdo.com/projects/story_permalink/1829486). The primary motivation is to centralize the location of all configuration default values and avoid repeating them or accidentally setting defaults to different values in different parts of the codebase.

## How Has This Been Tested?
I added an automated test for the new `config.get` behavior, and ensured that all tests continue to pass. Some tests and fixtures needed adjustment as they were implicitly relying on the in-code config defaults while not yet having loaded a default config.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -312,6 +314,8 @@ def stub_config():
     for key in default_keys:
         config.register(*key)
     config.extend(defaults.copy())
+    # Patch load() so we don't update any key/value pairs from actual files:
+    config.load = mock.Mock(side_effect=lambda: setattr(config, "ready", True))
```
Comment (jessesnyder) [inline tests/conftest.py]: I tried to figure out a clean way to replace the configuration wholesale in tests, or ... something else better than what we have, and failed. In the end I just moved the patching of `load` to the `stub_config` fixture and simplified `active_config` slightly, but it's essentially the same as what you had before.

File: tests/conftest.py [hunk 2]
```diff
@@ -322,12 +326,10 @@ def active_config(stub_config):
     """Loads the standard config as the active configuration returned by
     dallinger.config.get_config() and returns it.
     """
-    from dallinger.config import get_config
+    from dallinger import config as c
```
Comment (alecpm) [inline tests/conftest.py]: This is weird, I don't remember making this change!

File: dallinger/recruiters.py [hunk 3]
```diff
@@ -938,12 +940,12 @@ def from_config(config):
     the bot recruiter, which can be used in debug mode)
     and the MTurkRecruiter in other modes.
     """
-    debug_mode = config.get("mode", None) == "debug"
+    debug_mode = config.get("mode") == "debug"
```
[the remark this answers is not in the document]
Comment (alecpm) [reply dallinger/recruiters.py]: Mode is generally set by the experiment runner I think, so we can probably rely on it being set. I think I'd rather go back to having a default here than putting a default for mode in the config defaults, but I'm open to either.

File: tests/conftest.py (hunk 2 above)
Comment (alecpm) [reply tests/conftest.py]: Oh, @jessesnyder made that change on the original branch. Now it all makes sense.

File: dallinger/recruiters.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jessesnyder) [reply dallinger/recruiters.py]: I think if I was using the python API and didn't specify mode, I'd want it to try to run in debug mode. In general, I'd want to feel confident that things are going to be local/free by default.

[the remark this answers is not in the document]
Comment (alecpm) [reply dallinger/recruiters.py]: I've added `mode = debug` to the default config.

**Comment by jessesnyder** [comment]
@alecpm I think a rebase should fix Travis.

**Pull request merged**